content/guides/collaborations/connect-slack-to-group-collabs/4-handle-slack-events.md

```diff
@@ -0,0 +1,426 @@
+  team_id: 'T932DQSV12P',
+  team_domain: 'slacktest',
+  channel_id: 'C078N43MFHU',
+  channel_name: 'bottest',
+  user_id: 'U016JCDPN56',
+  user_name: 'testuser',
+  command: '/boxadd',
+  text: 'file 123456',
+  response_url: 'https://hooks.slack.com/commands/T541DQSV12P/3977594927231/ankvsRb42WKnKPRp002FeyTx',
+  trigger_id: '1189442196855.1183332180295.cca20c3ca1ea193dab432ad8e9e95431'
+}
+```
+
+  </Tab>
+  <Tab title='User Joined Channel Event'>
+
+```javascript
+{
+  token: 'cF1PwB1eIMcRHZWwFHJR1tgs',
+  team_id: 'T932DQSV12P',
+  api_app_id: 'A321V573PQT',
+  event: {
+    type: 'member_joined_channel',
+    user: 'U0431JM4RLZ',
+    channel: 'C078N43MFHU',
+    channel_type: 'C',
+    team: 'T932DQSV12P',
+    inviter: 'U016JCDPN56',
+    event_ts: '1592858788.000700'
+  },
+  type: 'event_callback',
+  event_id: 'Ev032NRJYASJ',
+  event_time: 1592858788,
+  authed_users: [ 'U0431JM4RLZ' ]
+}
+```
+
+  </Tab>
+  <Tab title='User Left Channel Event'>
+
+```javascript
+{
+  token: 'cF1PwB1eIMcRHZWwFHJR1tgs',
+  team_id: 'T932DQSV12P',
+  api_app_id: 'A321V573PQT',
+  event: {
+    type: 'member_left_channel',
+    user: 'U0431JM4RLZ',
+    channel: 'C078N43MFHU',
+    channel_type: 'C',
+    team: 'T932DQSV12P',
+    event_ts: '1593033236.000600'
+  },
+  type: 'event_callback',
+  event_id: 'Ev032NRJYASJ',
+  event_time: 1593033236,
+  authed_users: [ 'U0431JM4RLZ' ]
+}
+```
+
+  </Tab>
+</Tabs>
+
+<Choice option='programming.platform' value='node' color='none'>
+
+* Load `process.js` in your preferred editor.
+* In the `/event` listener, replace the `// HANDLE INCOMING EVENTS` comment with
+ the following.
+
+```javascript
+if (req.body.token !== config.verificationToken) {
+  res.send('Slack Verification Failed');
+}
+
+handler.process(res, req.body);
+```
+
+When an event comes through, the listener verifies that the message came from
+Slack, using the verification token from our Slack application. If it's a valid
+request, the event payload is sent to our event process function.
+
+</Choice>
+<Choice option='programming.platform' value='java' color='none'>
+
+```java
+
+```
+
+</Choice>
+<Choice option='programming.platform' value='dotnet' color='none'>
+
+```dotnet
+
+```
+
+</Choice>
+<Choice option='programming.platform' value='python' color='none'>
+
+```python
+
+```
+
+</Choice>
+<Choice option='programming.platform' value='ruby' color='none'>
+
+```ruby
+
+```
+
+</Choice>
+<Choice option='programming.platform' unset color='none'>
+  <Message danger>
+    # Incomplete previous step
+    Please select a preferred language / framework in step 1 to get started.
+  </Message>
+</Choice>
+
+## Process Slack events
+
+<Choice option='programming.platform' value='node' color='none'>
+
+In the `process` function, replace the `// PROCESS EVENTS` comment with the
+following.
+
+<!-- markdownlint-disable line-length -->
+```javascript
+let userId;
+
+if (
+  data.type && 
+  data.type === 'event_callback'
```

Comment:
No need for this double check. `data.type === 'event_callback` will do, cause we know `data` is present.